### PR TITLE
Add logical operations: Nand and Nor.

### PR DESCRIPTION
**Pull request by rocky**
Note: there will be futher changes to And when we merge ConditionalExpressions.

### CODE REVIEW

**Comment by rocky** [comment]
@mmatera when this clears and gets into master, I will rebase https://github.com/mathics/Mathics/pull/1330 for the further changes that are needed there. I can either put this in a new branch, or overwrite the existing one - your choice. 

But the idea here, is that `Nand` and `Nor` is a separate useful thing independent of Condtional Expressions. By having these separated it helps better understand what's going on not just in that PR but in the future if we try to understand what's up or decouple the two.

Thanks for your patience and understanding.

**Comment by mmatera** [comment]
@rocky, thanks for doing this. Go ahead with the merging and rebasing.

File: mathics/builtin/logic.py [hunk 1]
```diff
@@ -118,6 +125,38 @@ class Not(PrefixOperator):
     }
 
 
+class Nand(Builtin):
+    """
+    <dl>
+    <dt>'Nand[$expr1$, $expr2$, ...]'
+    <dt>'$expr1$ && $expr2$ && ...'
+        <dd> Implements the logical `NOR` function.  The same that Not[And[$expr1$, $expr2$, ...]]
```
Comment (axkr) [inline mathics/builtin/logic.py:133]: NOR seems to be a typo.

Comment (rocky) [reply mathics/builtin/logic.py:133]: Thanks - should be fixed now in 369f6d18

**Pull request merged**